File: jdba-core/src/main/java/org/jetbrains/jdba/core/DBCommandRunner.java
package org.jetbrains.jdba.core;

/**
 * @author Leonid Bushuev from JetBrains
 */
public interface DBCommandRunner {

  /**
   * Assigns command parameters.
   * @param params parameters, one for each '?' sign in the command.
   * @return the query.
   */
  DBCommandRunner withParams(Object... params);

  /**
   * Performs the query.
   */
  DBCommandRunner run();
}

File: DekafMain/src/core/DBCommandRunner.kt
package org.jetbrains.dekaf.core

/**
 * @author Leonid Bushuev
 */
interface DBCommandRunner {

    /**
     * Assigns command parameters.
     * @param params parameters, one for each '?' sign in the command.
     * *
     * @return the query.
     */
    fun withParams(vararg params: Any): DBCommandRunner

    /**
     * Performs the query.
     */
    fun run(): DBCommandRunner
    
}

